### PR TITLE
Start migration of WebDev to null-safety

### PR DESCRIPTION
**Pull request by elliette**
Work towards https://github.com/dart-lang/webdev/issues/1640

### CODE REVIEW

File: webdev/lib/src/pubspec.dart [hunk 1]
```diff
@@ -20,18 +17,18 @@ import 'version.dart';
...
 
 class PackageExceptionDetails {
   final String error;
-  final String description;
+  final String? description;
   final bool _missingDependency;
 
   const PackageExceptionDetails._(this.error,
-      {this.description, bool missingDependency})
+      {this.description, bool? missingDependency})
```
Comment (annagrin) [inline webdev/lib/src/pubspec.dart:31]: can this be non-null with default value "false" instead?

File: webdev/lib/src/pubspec.dart [hunk 2]
```diff
@@ -82,28 +79,31 @@ class PubspecLock {
...
 
-    var packages = pubspecLock['packages'] as YamlMap;
+    var packages = pubspecLock['packages'] as YamlMap?;
     return PubspecLock(packages);
   }
 
   List<PackageExceptionDetails> checkPackage(
       String pkgName, VersionConstraint constraint,
-      {String forArgument, bool requireDirect}) {
+      {String? forArgument, bool? requireDirect}) {
```
Comment (annagrin) [inline webdev/lib/src/pubspec.dart:88]: Can this be non-null with default value "false"?

File: webdev/pubspec.yaml [hunk 3]
```diff
@@ -49,9 +49,9 @@ dev_dependencies:
   webdriver: ^3.0.0
 
 # Comment out before releasing webdev.
-# dependency_overrides:
-#   dwds:
-#     path: ../dwds
+dependency_overrides:
```
Comment (annagrin) [inline webdev/pubspec.yaml:52]: We also should update the webdev version in this file and in changelog.md to a '-dev' one.

**Review by annagrin — COMMENTED** [review]
Thanks Elliott! I left a few comments.

File: webdev/lib/src/pubspec.dart (hunk 1 above)
Comment (elliette) [reply webdev/lib/src/pubspec.dart]: Done!

File: webdev/pubspec.yaml (hunk 3 above)
Comment (elliette) [reply webdev/pubspec.yaml:52]: Done

**Review by annagrin — APPROVED** [review]
Thanks Elliott, LGTM!

**Pull request merged**